### DISCUSSION TITLE
[Feature Request]  Code editor box with syntax highlighting

### DISCUSSION THREAD

**Discussion by AtharvaKolekar** [heroui-inc/heroui#4723] [Feature requests]
### Is your feature request related to a problem? Please describe.

I was looking for UI component for code-box https://nextui.org/docs/components/textarea but didn't found it.

### Describe the solution you'd like

A new component I want to suggest is code editor similar to prism-code, which highlights syntax to desired language.
It can be flexible (edit or read-only), copy button, language selector, custom themes, numbering the lines, external link to execute the code......just how the docs page have the code tab option.

### Describe alternatives you've considered

prism-code

### Screenshots or Videos

![image](https://github.com/user-attachments/assets/03e5e2a6-b2c0-4015-b4a7-1ee199037101)
Example is itself inside the docs

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1535/[feature-request]-code-editor-box-with-syntax-highlighting">ENG-1535 [Feature Request] Code editor box with syntax highlighting</a></p>
